src/main/java/com/zendesk/maxwell/replication/BinlogConnectorReplicator.java

```diff
@@ -41,11 +41,14 @@
 	private static final long MAX_TX_ELEMENTS = 10000;
 	public static final int BAD_BINLOG_ERROR_CODE = 1236;
 	public static final int ACCESS_DENIED_ERROR_CODE = 1227;
+	public static final int HEARTBEAT_FAILURE_ERROR_CODE = 1623;
 
 	private final String clientID;
 	private final String maxwellSchemaDatabaseName;
 
 	protected final BinaryLogClient client;
+	private final long heartbeatInterval = 5000L; // TODO configure
+	private final float heartbeatIntervalAllowance = 3.0f; // TODO configure, find a better name?
```

Comment:
I'd say let's not configure these, but instead just have a boolean flag that turns on/off the master heartbeating feature.  1 second  Interval and 5 second allowance seem fine?

Reply:
Sure, less work for me 👍

Reply:
One thing to point out: the implementation treats "construction of this java object" as the first heartbeat, so it incidentally includes connection time in the first heartbeat. But we hardcode connection timeout to 5s as well anyway, so this shouldn't be an issue.

Reply:
good to know you're thinking of the corner cases.  I too was wondering if there were any places where we might timeout inelegantly waiting for the connection to come up or something. 

what does happen with a timing out server, I wonder.